### PR TITLE
Add type hints to L2CAP module

### PR DESCRIPTION
**Pull request by zxzxwu**
Also slightly refactor L2CAP channel tables. The original implementation mix BR/EDR and LE channels.

### CODE REVIEW

**Comment by zxzxwu** [comment]
This work is much larger than I expected...

**Comment by uael** [comment]
We don't need to do the whole thing at once, a subset is good enough.
But I'm wondering if it's worth introducing breaking changes by refactoring at the same time? Maybe we should just add some typing at first, what do you think?

**Comment by zxzxwu** [comment]
Right, let me think how to minimize this

**Comment by zxzxwu** [comment]
I removed the refactor of code enum, and avoid any hints related to logic change.

File: bumble/l2cap.py [hunk 1]
```diff
@@ -256,29 +271,29 @@ def inner(cls):
...
         if hasattr(self, 'fields') and kwargs:
             HCI_Object.init_from_fields(self, self.fields, kwargs)
-        if pdu is None:
-            data = HCI_Object.dict_to_bytes(kwargs, self.fields)
-            pdu = (
-                bytes([self.code, self.identifier])
-                + struct.pack('<H', len(data))
-                + data
-            )
+            if pdu is None:
```
Comment (uael) [inline bumble/l2cap.py:278]: `if hasattr(self, 'fields') and pdu is None:` and keep depth as is ?

**Review by uael — APPROVED** [review]

File: bumble/l2cap.py (hunk 1 above)
Comment (zxzxwu) [reply bumble/l2cap.py:278]: Oh I missed kwarg. How about
```py
def __init__(self, pdu=None, **kwargs) -> None:
        self.identifier = kwargs.get('identifier', 0)
        if hasattr(self, 'fields'):
            if kwargs:
                HCI_Object.init_from_fields(self, self.fields, kwargs)
            if pdu is None:
                data = HCI_Object.dict_to_bytes(kwargs, self.fields)
                pdu = (
                    bytes([self.code, self.identifier])
                    + struct.pack('<H', len(data))
                    + data
                )
        self.pdu = pdu
```

Comment (uael) [reply bumble/l2cap.py:278]: lgtm!

**Pull request merged**